orderbook/src/orderbook.rs

```diff
@@ -85,6 +91,48 @@ impl OrderBook {
         orders.retain(|_, order| has_future_valid_to(now_in_epoch_seconds, &order.order_creation));
     }
 
+    async fn remove_settled_orders(&self, settlement_contract: &GPv2Settlement) {
+        let uids_to_be_removed = self.get_uids_of_settled_orders(settlement_contract).await;
+        info!(
+            "removing following settled orders from orderbook: {:?}",
+            uids_to_be_removed
+        );
+        let mut orders = self.orders.write().await;
+        orders.retain(|uid, _| uids_to_be_removed.contains(uid));
+    }
+
+    async fn get_uids_of_settled_orders(
+        &self,
+        settlement_contract: &GPv2Settlement,
+    ) -> Vec<OrderUid> {
+        let orders = self.orders.read().await;
+        let uid_futures = orders.iter().map(|(uid, _)| async move {
+            self.has_uid_been_settled(*uid, settlement_contract).await
+        });
+        let uid_pairs: Vec<Option<OrderUid>> = join_all(uid_futures).await;
```

Comment:
Have to be a little careful here since all futures run in parallel and there might be a number of connections limit on the node. Doing sequentially doesn't have this problem but is slower. I feel like sequentially might be safer until we use a batched transport.

Reply:
Yeah, maybe it is safer. Though, I don't think this is really a risk for our own nodes. They are setup with the following config:
```
            "--tx-queue-size=8192",
```
Assuming one call does not take longer than 1 second to work on, this setting allows us to easily have more than 8000 requests per second

Reply:
If that does indeed control the number of connections then it's probably fine.